internal/config/dependencies.go

```diff
@@ -0,0 +1,22 @@
+package config
+
+import (
+	"github.com/go-shiori/shiori/internal/database"
+	"github.com/go-shiori/shiori/internal/domains"
+	"go.uber.org/zap"
+)
+
+type Dependencies struct {
```

Comment:
I don't think it's proper to put this `Dependencies` here under the `config` dir. And I think it's a little early to put dependencies in a struct. How about we do these things later when enough dependencies occur?

Reply:
Yeah, when there is many dependencies, it's better to do like this. I just think maybe it can be added in the future when really needed.Actually I mean if the signature like this:
```
func NewAuthAPIRoutes(logger *zap.Logger, db database.DB, authDomain domains.AuthDomain) *AuthAPIRoutes {
```
it seems more straightforward since there are just two dependencies for AuthAPI.

Reply:
Yeah, maybe I tried looking too far into the future. I will change it to your approach using the simpler dependency injection pattern.